app/modules/l3g4200d.c

```diff
@@ -0,0 +1,104 @@
+/*
+ * Driver for STM L3G4200D three axis gyroscope
+ *
+ * Code based on BMP085 driver.
+ */
+#include "module.h"
+#include "lauxlib.h"
+#include "platform.h"
+#include "c_stdlib.h"
+#include "c_string.h"
+
+static const uint32_t i2c_id = 0;
+static const uint8_t i2c_addr = 0x69;
+
+static uint8_t ICACHE_FLASH_ATTR r8u(uint32_t id, uint8_t reg) {
+    uint8_t ret;
+
+    platform_i2c_send_start(id);
+    platform_i2c_send_address(id, i2c_addr, PLATFORM_I2C_DIRECTION_TRANSMITTER);
+    platform_i2c_send_byte(id, reg);
+    platform_i2c_send_stop(id);
+    platform_i2c_send_start(id);
+    platform_i2c_send_address(id, i2c_addr, PLATFORM_I2C_DIRECTION_RECEIVER);
+    ret = platform_i2c_recv_byte(id, 0);
+    platform_i2c_send_stop(id);
+    return ret;
+}
+
+static void ICACHE_FLASH_ATTR w8u(uint32_t id, uint8_t reg, uint8_t val) {
+    platform_i2c_send_start(i2c_id);
+    platform_i2c_send_address(i2c_id, i2c_addr, PLATFORM_I2C_DIRECTION_TRANSMITTER);
+    platform_i2c_send_byte(i2c_id, reg);
+    platform_i2c_send_byte(i2c_id, val);
+    platform_i2c_send_stop(i2c_id);
+}
+
+static int ICACHE_FLASH_ATTR l3g4200d_init(lua_State* L) {
+
+    uint32_t sda;
+    uint32_t scl;
+    uint8_t  devid;
+
+    if (!lua_isnumber(L, 1) || !lua_isnumber(L, 2)) {
+        return luaL_error(L, "wrong arg range");
+    }
+
+    sda = luaL_checkinteger(L, 1);
+    scl = luaL_checkinteger(L, 2);
+
+    if (scl == 0 || sda == 0) {
```

Comment:
Use a luaL_argcheck() for these tests.  The code is shorter and the error message more informative.